File: Medical_Profile/extension.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Windows.Forms;

namespace Medical_Profile
{
 static class ControlExtensions
 {
  private readonly static int WM_SETREDRAW = 11;

  public static void SuspendPaint(this Control ctrl)
  {
   var msgSuspendUpdate = Message.Create(ctrl.Handle, WM_SETREDRAW, IntPtr.Zero, IntPtr.Zero);
   var window = NativeWindow.FromHandle(ctrl.Handle);
   window.DefWndProc(ref msgSuspendUpdate);
  }

  /// <summary>
 /// Resume from SuspendPaint method
 /// </summary>
 /// <param name="ctrl"></param>
 /// <remarks></remarks>
  public static void ResumePaint(this Control ctrl)
  {
   var wparam = new IntPtr(1);
   var msgResumeUpdate = Message.Create(ctrl.Handle, WM_SETREDRAW, wparam, IntPtr.Zero);
   var window = NativeWindow.FromHandle(ctrl.Handle);
   window.DefWndProc(ref msgResumeUpdate);
   ctrl.Refresh();
  }

  public static KeyValuePair<TKey, TValue>? Getlkvp<TKey, TValue>(this SortedDictionary<TKey, TValue> sdictionary)
  {
   if (sdictionary.Keys.Count == 0)
   {
    return default;
   }

   int ct = sdictionary.Count - 1;
   var k = sdictionary.Keys.ElementAtOrDefault(ct);
   var v = sdictionary[k];
   return new KeyValuePair<TKey, TValue>(k, v);
  }

  public static TKey Getlkey<TKey, TValue>(this SortedDictionary<TKey, TValue> sdictionary)
  {
   if (sdictionary.Keys.Count == 0)
   {
    return default;
   }

   int ct = sdictionary.Count - 1;
   var k = sdictionary.Keys.ElementAtOrDefault(ct);
   var v = sdictionary[k];
   return k;
  }

  public static TValue Getlval<TKey, TValue>(this SortedDictionary<TKey, TValue> sdictionary)
  {
   if (sdictionary.Keys.Count == 0)
   {
    return default;
   }

   int ct = sdictionary.Count - 1;
   var k = sdictionary.Keys.ElementAtOrDefault(ct);
   var v = sdictionary[k];
   return v;
  }

  public static void CenterControl(this Control ctrl)
  {
   ctrl.Top = (int)(ctrl.Parent.Height / (double)2 - ctrl.Height / (double)2);
   ctrl.Left = (int)(ctrl.Parent.Width / (double)2 - ctrl.Width / (double)2);
  }
 }
}

File: Medical_Profile/extension.vb
Imports System.Runtime.CompilerServices
Imports System.Text

Module ControlExtensions
  ReadOnly WM_SETREDRAW As Integer = 11

  <Extension()>
 Public Sub SuspendPaint(ByVal ctrl As Windows.Forms.Control)

  Dim msgSuspendUpdate As Windows.Forms.Message = Windows.Forms.Message.Create(ctrl.Handle, WM_SETREDRAW, System.IntPtr.Zero, System.IntPtr.Zero)

  Dim window As Windows.Forms.NativeWindow = Windows.Forms.NativeWindow.FromHandle(ctrl.Handle)

  window.DefWndProc(msgSuspendUpdate)

 End Sub

 ''' <summary>
 ''' Resume from SuspendPaint method
 ''' </summary>
 ''' <param name="ctrl"></param>
 ''' <remarks></remarks>
 <Extension()>
 Public Sub ResumePaint(ByVal ctrl As Windows.Forms.Control)

  Dim wparam As New System.IntPtr(1)
  Dim msgResumeUpdate As Windows.Forms.Message = Windows.Forms.Message.Create(ctrl.Handle, WM_SETREDRAW, wparam, System.IntPtr.Zero)

  Dim window As Windows.Forms.NativeWindow = Windows.Forms.NativeWindow.FromHandle(ctrl.Handle)

  window.DefWndProc(msgResumeUpdate)

  ctrl.Refresh()
 End Sub


  <Extension()>
 Public Function Getlkvp(Of TKey, TValue)(sdictionary As SortedDictionary(Of TKey, TValue)) As KeyValuePair(Of TKey, TValue)?
  If sdictionary.Keys.Count = 0 Then
   Return Nothing
  End If
  Dim ct As Integer = sdictionary.Count - 1
  Dim k As TKey = sdictionary.Keys(ct)
  Dim v As TValue = sdictionary(k)
  Return New KeyValuePair(Of TKey, TValue)(k, v)
 End Function


 <Extension()>
 Public Function Getlkey(Of TKey, TValue)(sdictionary As SortedDictionary(Of TKey, TValue)) As TKey
  If sdictionary.Keys.Count = 0 Then
   Return Nothing
  End If
  Dim ct As Integer = sdictionary.Count - 1
  Dim k As TKey = sdictionary.Keys(ct)
  Dim v As TValue = sdictionary(k)
  Return k
 End Function

 <Extension()>
 Public Function Getlval(Of TKey, TValue)(sdictionary As SortedDictionary(Of TKey, TValue)) As TValue
  If sdictionary.Keys.Count = 0 Then
   Return Nothing
  End If
  Dim ct As Integer = sdictionary.Count - 1
  Dim k As TKey = sdictionary.Keys(ct)
  Dim v As TValue = sdictionary(k)
  Return v
 End Function

  <Extension()>
  Public Sub CenterControl(ByVal ctrl As Windows.Forms.Control)
    ctrl.Top = (ctrl.Parent.Height / 2) - (ctrl.Height / 2)
    ctrl.Left = (ctrl.Parent.Width / 2) - (ctrl.Width / 2)
  End Sub
End Module
